### ISSUE TITLE
Spring REST Client generation

### ISSUE THREAD

**Issue by raman-savitski** [vojtechhabarta/typescript-generator#488]
Currently only `@RestController` annotation is used to detect classes which should be generated as REST Clients.
I'd like to see same behavior for `@RequestMapping` annotation. In my application there are lots of REST controllers which use only this annotation and currently there are no possibility to generate REST Clients for them.

For example:
```
@RequestMapping("/rest/user")
public static class UserController {
       @RequestMapping("/{id}")
       public User getUser() {
            .....
       }
}
```

Can I submit pull request with this changes?

`SpringApplicationParser.java:115`
```
        // controller
        final RestController controller = AnnotationUtils.findAnnotation(cls, RestController.class);
        final RequestMapping mapping = AnnotationUtils.findAnnotation(cls, RequestMapping.class);
        if (controller != null || mapping != null) {
             .....
```

**Comment by vojtechhabarta** [comment]
How does it work in your case? When I remove `@RestController` annotation from some class Spring doesn't consider it being a controller anymore (even if it has `@RequestMapping`). Do you have your controllers specified in some XML file?

**Comment by raman-savitski** [comment]
Sorry for misleading. Actual issues is not about `@RequestMapping` annotation.
We're using multiple annotations to mark spring beans. `@Controller` `@Service` `@RestController`. I'd like the ability to specify annotations which should be treated as  "generate REST client for this class".

**Comment by vojtechhabarta** [comment]
Ok, understood.

I think typescript-generator should be reasonably consistent with Spring. And if I understand it correctly `@RequestMapping` annotation alone (without `@RestController` or other annotation) is not REST controller. So the question is how we should reliably detect REST controllers. `@RestController` annotation is itself annotated with `@Controller`. Furthermore `@Controller` and `@Service` are annotated with `@Component`. So maybe we should use all annotations that are transitively annotated with `@Component`. Or maybe we should just use predefined list of annotations. What do you think?

**Comment by raman-savitski** [comment]
I think `@RestController` `@Controller` `@Service` should cover all reasonable cases.

**Comment by raman-savitski** [comment]
Yep, works as expected. Thank you.

**Comment by vojtechhabarta** [comment]
Released in [2.24.612](https://github.com/vojtechhabarta/typescript-generator/releases/tag/v2.24.612).
